inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -24,8 +25,12 @@ def __init__(self, db_file=UDM_DB_FILE) -> None:
         """
         self._db_file = db_file
         if self._db_file != ":memory:" and not os.path.exists(self._db_file):
-            raise DispatcherException(f"Error: Database file {self._db_file} does not exist.")
-
+            # Create database file if not exist
+            logger.info(f"Database file doesn't exist. Creating the file.")
+            file_descriptor = os.open(self._db_file, os.O_CREAT)
+            # Set permission of the file (rw for owner and group)
+            os.chmod(self._db_file, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IWGRP)
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Ensure that sensitive operations such as file creation and database transactions are securely handled.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            if not os.path.exists(self._db_file):
                file_descriptor = os.open(self._db_file, os.O_CREAT | os.O_EXCL)
                os.chmod(self._db_file, stat.S_IRUSR | stat.S_IWUSR)
                os.close(file_descriptor)
```